### PR TITLE
Move the code for Integrations from TEC to Common

### PR DESCRIPTION
**Pull request by bordoni**
🎫 [ET-1552](https://theeventscalendar.atlassian.net/browse/ET-1552) [ET-1553](https://theeventscalendar.atlassian.net/browse/ET-1553)
---
Added DevDocs documentation about this:
https://docs.theeventscalendar.com/apis/integrations/including-new-integrations/

[ET-1552]: https://theeventscalendar.atlassian.net/browse/ET-1552?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ET-1553]: https://theeventscalendar.atlassian.net/browse/ET-1553?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by stratease** [comment]
Were there any existing tests that could be moved here? Could test with dynamic classes if nothing exists already.

File: src/Common/Integrations/Integration_Abstract.php [hunk 1]
```diff
@@ -0,0 +1,128 @@
+<?php
+/**
+ * Abstract for Integrations.
+ *
+ * @since   TBD
```
Comment (rafsuntaskin) [inline src/Common/Integrations/Integration_Abstract.php:5]:
```suggestion
 * @since TBD
```

Comment (rafsuntaskin) [reply src/Common/Integrations/Integration_Abstract.php:5]: Needs a bulk search-replace to get rid of all this kind of extra spaces in multiple places.

**Comment by bordoni** [comment]
> Were there any existing tests that could be moved here? Could test with dynamic classes if nothing exists already.

100% @stratease it's missing tests for now, getting back to that later. Just wanted to see if anyone had things to add to the implementation of the integrations structure.

**Review by Camwyn — DISMISSED** [review]
Looks great - but see comments on https://github.com/the-events-calendar/the-events-calendar/pull/4196 for possible implications.

**Review by stratease — DISMISSED** [review]

**Pull request merged**